app.rb

```diff
@@ -0,0 +1,167 @@
+require_relative './book'
+require_relative './person'
+require_relative './rental'
+require_relative './student'
+require_relative './teacher'
+
+class App
+  def initialize
+    @books = []
+    @rentals = []
+    @person = []
+  end
+
+  def menu
+    puts 'Welcome to the Library'
+    puts 'Please choose an option by enterung a number:'
+    puts '1 - List all books'
+    puts '2 - List all People'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Createe a rental'
+    puts '6 - List all rentals for a given person id'
+    puts '7 - Exit'
+    gets.chomp
+  end
+
+  def menu_option
+    choice = menu.to_s
+    case choice
+    when '1'
+      list_books
+    when '2'
+      list_persons
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_all_rentals_for_a_given_person_id
+    else
+      puts 'Thank you for using this app'
+      exit
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    input1 = gets.chomp
+    print 'Author: '
+    input2 = gets.chomp
+    book = Book.new(input1, input2)
+    @books.push(book)
+    puts 'Book created successfully '
+    menu_option
+  end
+
+  def create_person
+    print "\n Do you want to create a student (1) or a teacher (2)? [Input the number]:"
+    user_input = gets.chomp
+    case user_input
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      print 'Please input either 1 or 2'
+      create_person
+    end
+    puts 'Person created successfully'
+    gets
+    menu_option
+  end
+
+  def create_rental
+    print 'Select a book'
+    list_all_books
+    input1 = gets.chomp.to_i
+    print 'Select a person'
+    list_all_people
+    input2 = gets.chomp.to_i
+    print "\n Date(yyyy/mm/dd): "
+    date = gets.chomp.to_i
+    rental = Rental.new(date, @person[input2], @books[input1])
+    @rental.push(rental)
+    puts 'Rental created successfully'
+    menu_option
+  end
+
+  def list_all_books
+    @books.each_with_index do |item, index|
+      puts "#{index} Title: \"#{item.title}\", Author: #{item.author}"
+    end
+  end
+
+  def list_books
+    list_all_books
+    puts 'Press enter to continue.'
+    gets.chomp
+    menu_option
+  end
+
+  def list_all_people
+    @person.each_with_index do |item, index|
+      puts "#{index} Name: #{item.name} Age: #{item.age} ID: #{item.id}"
+    end
+  end
+
+  def list_persons
+    list_all_people
+    puts 'Please Enter to continue'
+    gets.chomp
+    menu_option
+  end
```

Comment:
- [ ] I noticed that when I choose a number may be `list of books` you have to repeat it twice to work check here I created students and after I wanted to check `list of them` and I had to repeat 2 twice, please make sure anytime I choose one option to work without repeating it
![bk](https://user-images.githubusercontent.com/44745855/187951007-ca82c407-27b1-4d03-b3e9-1b21916c91f7.png)

Reply:
Implemented 👍